doc: relnotes: Document "other" section for Zephyr upmerge (#3893)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

  * Removed the legacy DT macros.
  * Started exposing dependency ordinals for walking the dependency hierarchy.
  * Added documentation for the :ref:`DTS bindings <zephyr:devicetree_binding_index>`.

>   * Added the ``UICR`` and ``FICR`` peripherals to the Devicetree.

Reviewer: ```suggestion
  * Added the ``UICR`` and ``FICR`` peripherals to the devicetree.
```

>   * Interrupt priorities in Devicetree on Nordic devices now default to ``NRF_DEFAULT_IRQ_PRIORITY`` instead of hard-coded values.

Reviewer: ```suggestion
  * Changed the interrupt priorities in devicetree for Nordic Semiconductor devices to default to ``NRF_DEFAULT_IRQ_PRIORITY`` instead of hard-coded values.
```

      See :option:`CONFIG_MCUBOOT_SHELL`.
    * Reworked the implementation to use MCUboot's bootutil_public library instead of the Zephyr implementation of the same API.

  * IPC:

>     * Added a new ``subsys/ipc`` subsystem that provides multi-endpoint capabilites to the OpenAMP integration.

Reviewer: ```suggestion
    * Added a ``subsys/ipc`` subsystem that provides multi-endpoint capabilities to the OpenAMP integration.
```

  * Updated LVGL to v7.6.1.
  * Updated libmetal and openamp to v2020.10.
  * Updated nrfx in hal-nordic to version 2.4.0.

>   * Moved the nrfx glue code from the hal_nordic module repository to the main zephyr tree.

Reviewer: ```suggestion
  * Moved the nrfx glue code from the hal_nordic module repository to the main Zephyr tree.
```

Reviewer 2: @ru-fu the upstream Zephyr project convention is to use lowercase 'z' when referring to the Git repository whose name is "zephyr", as is being done here ("tree" means "Git repository"). Does NCS have the opposite convention? That should probably be reversed if so, IMO.

Reviewer: In NCS context, the repo is usually sdk-zephyr, so no need for this kind of convention. I was considering if this refers to "Zephyr" in general or to the repo and figured if it was the repository, it should say repository and not tree. ;)
But I'm okay with leaving it lowercase.

Reviewer 2: OK, I understand.

The repository's name in west.yml is still 'zephyr', and its path is 'zephyr', so the fact that its current URL contains 'sdk-zephyr' might not be something users are familiar with, I think. So I think that between that and keeping convention with the  zephyr docs, a lowercase 'z' may be best since you are agreeable.

Author: I do think "zephyr tree" is an upstream term that we should perhaps avoid. Instead we could potentially refer to it as the "main Zephyr repository". I will make that change.

  * Updated the Trusted Firmware-M (TF-M) module to include support for the nRF5340 and nRF9160 platforms.

* Other:

  * Renamed the ``sanitycheck`` script to ``twister``.
  * Added initial LoRaWAN support.
  * Updated ``west flash`` support for ``nrfjprog`` to fail if a HEX file has UICR data and ``--erase`` was not specified.

>   * Added a new API to correlate system time with external time sources (see :ref:`zephyr:timeutil_api`).

Reviewer: ```suggestion
  * Added an API to correlate system time with external time sources (see :ref:`zephyr:timeutil_api`).
```

* Power management:

  * Overhauled the naming and did some general cleanup.

>   * Added a new notifier API to register an object to receive notifications when the system changes power state.

Reviewer: ```suggestion
  * Added a notifier API to register an object to receive notifications when the system changes power state.
```
